apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -1279,6 +1284,34 @@ const (
 	SingleContainerStrategy ContainerStrategyType = "single"
 )
 
+// +k8s:openapi-gen=true
+type UpdateStrategy struct {
+	// Type can be "RollingUpdate" or "OnDelete" for DaemonSets and "Rolling Update"
+	// "Recreate" for Deployments
+	Type string `json:"type,omitempty"`
+	// Configure the rolling update strategy of the Deployment or DaemonSet.
+	RollingUpdate *RollingUpdate `json:"rollingUpdate,omitempty"`
+}
+
+// Describes how to replace existing pods with new ones.
+// +k8s:openapi-gen=true
+type RollingUpdate struct {
+	// The maximum number of pods that can be unavailable during the update.
+	// Value can be an absolute number (ex: 5) or a percentage of desired pods (ex: 10%).
+	// Absolute number is calculated from percentage by rounding down.
```

Comment:
The daemonset docs use `rounding up`, so maybe we can remove this line or clarify that daemonset is round up and deployment is round down